src/sorting/mod.rs

```diff
@@ -98,6 +166,10 @@ mod tests {
         let mut singleton_array: Vec<i32> = vec![42];
         sorting_function(&mut singleton_array);
         assert_eq!(singleton_array, vec![42]);
+
+        let mut longer_list = vec![4, 7, 11, 0, 3, 5, 42, 12];
```

Comment:
Can we also have a test with duplicated items?

And possibly also one with all items equal to each other (e.g. `[0, 0, 0, 0]`).

The first would test partition correctness in presence of duplicates, the second would test edge cases in pointers crossing in Hoare partitioning scheme.